extensions/wikia/AdEngine/js/recovery/message.js

```diff
@@ -14,14 +20,54 @@ define('ext.wikia.adEngine.recovery.message', [
 
 	var logGroup = 'ext.wikia.adEngine.recovery.message',
 		wikiaTopAdsId = 'WikiaTopAds',
-		wikiaRailId = 'WikiaRail';
+		wikiaRailId = 'WikiaRail',
+		localstorageKey = 'rejectedRecoveredMessage',
+		headerText = 'Hey! It looks like you\'re using ad blocking software!',
+		messageText = 'Wikia runs ads so we can keep the lights on and so Wikia will always be free to use. ' +
+			'We can bring you fun, free, fan-oriented content until my glorious return to Earh if you ' +
+			'<strong>add us to your adblock whitelist</strong>. ' +
+			'<a class="action-accept">Click</a> my face to refresh after you\'re done!';
+
+	function accept() {
+		adTracker.track('recovery/message', 'accept');
+		location.reload();
+	}
+
+	function reject(message) {
+		adTracker.track('recovery/message', 'reject');
+		message.style.display = 'none';
+		localStorage.setItem(localstorageKey, true);
+	}
+
+	function isRejected() {
+		return !!localStorage.getItem(localstorageKey);
+	}
 
 	function createMessage(uniqueClassName) {
 		var className = 'recovered-message',
-			div = doc.createElement('div');
+			div = doc.createElement('div'),
```

Comment:
I think at this point it's better to move everything to a mustache template and request it. We're starting to have more and more spaghetti here ;) Moving it to a template comes with performance bonus points, I guess: the template response can be cached :)